### ISSUE TITLE
HAMi multi-machine multi-GPU distributed training, GPU memory exception error

### ISSUE THREAD

**Issue by 2232729885** [Project-HAMi/HAMi#1118]
**What happened**:
我在k8s集群（v1.27.2）中部署了kubeflow-training-operator（v1.8.0）和hami-2.6.0，我的集群中有两个gpu节点，每个节点都有8张A40卡。当我创建一个pytorch ddp分布式训练任务（一个很简单的ddp demo，一个master pod， 一个worker pod，每个pod分配8个vgpu，每个vgpu分配2g显存）时,master pod 报如下图错误，显示卡5需要使用远超过我分配的2g显存。但是我在物理机执行 nvidia-smi，卡5还有10多g空闲显存，并且ddp demo只需要几百M显存就可以了，这显然不合理。

![Image](https://github.com/user-attachments/assets/0644db46-2af6-4561-9cc8-8dbd4e5035fa)

机器上执行nvidia-smi的结果：

![Image](https://github.com/user-attachments/assets/63985ddb-c7a9-47c9-9153-7d7d26d0deb8)
master pod内刚开始执行nvidia-smi的结果：

![Image](https://github.com/user-attachments/assets/e496da72-6ba0-4e44-832d-48ac479c6a4b)

pytorch ddp demo：
`import torch
import torch.distributed as dist
import torch.nn as nn
import torch.optim as optim
import os

from torch.nn.parallel import DistributedDataParallel as DDP

class ToyModel(nn.Module):
    def __init__(self):
        super(ToyModel, self).__init__()
        self.net1 = nn.Linear(10, 10)
        self.relu = nn.ReLU()
        self.net2 = nn.Linear(10, 5)

    def forward(self, x):
        return self.net2(self.relu(self.net1(x)))

def demo_basic():
    torch.cuda.set_device(int(os.environ["LOCAL_RANK"]))
    dist.init_process_group("nccl")
    rank = dist.get_rank()

    # 打印 rank 和当前 GPU
    cur_dev = torch.cuda.current_device()
    print(f"[Rank {rank}] running on GPU {cur_dev}")
    # print(f"Start running basic DDP example on rank {rank}.")

    # create model and move it to GPU with id rank
    device_id = rank % torch.cuda.device_count()
    model = ToyModel().to(device_id)
    ddp_model = DDP(model, device_ids=[device_id])
    loss_fn = nn.MSELoss()
    optimizer = optim.SGD(ddp_model.parameters(), lr=0.001)

    # 再次确认前向时的设备
    print(f"[Rank {rank}] model and inputs on GPU {device_id}")

    optimizer.zero_grad()
    outputs = ddp_model(torch.randn(20, 10))
    labels = torch.randn(20, 5).to(device_id)
    loss = loss_fn(outputs, labels)
    loss.backward()
    optimizer.step()

    print(f"[Rank {rank}] step done on GPU {device_id}, loss={loss.item():.4f}")

    # 在此处打印参数，验证同步
    for name, param in ddp_model.named_parameters():
        print(f"[Rank {rank}] {name}: {param.data.view(-1)[0].item()} ")

    dist.destroy_process_group()

    print(f"Finished running basic DDP example on rank {rank}.")

if __name__ == "__main__":
    demo_basic()`

pod 训练命令：torchrun --nnodes=2 --nproc_per_node=8 elastic_ddp.py

**What you expected to happen**:
hami 显存使用不报错，日志显示需要利用超大的显存显然不对

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

- The output of `nvidia-smi -a` on your host
-
- Your docker or containerd configuration file (e.g: `/etc/docker/daemon.json`)
- The hami-device-plugin container logs
- The hami-scheduler container logs
- The kubelet logs on the node (e.g: `sudo journalctl -r -u kubelet`)
- Any relevant kernel output lines from `dmesg`

**Environment**:
- HAMi version: 2.6.0
- nvidia driver or other AI device driver version: 535.104.05
- Docker version from `docker version`：26.1.4
- Docker command, image and tag used 
- Kernel version from `uname -a` Linux node1 3.10.0-1160.119.1.el7.x86_64 #1 SMP Tue Jun 4 14:43:51 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
- Others:

**Comment by 2232729885** [comment]
补充：我在pod内少起几个训练进程就没报错，比如每个pod分配 4个vgpu，训练命令使用torchrun --nnodes=2 --nproc_per_node=4 elastic_ddp.py 就不会报错，但是pod分配多个vgpu，比如8个就报错

**Comment by 2232729885** [comment]
> 你试试 `kubectl describe nodes` 看看 Allocatable: gpu 有几个？再看看 Allocated resources: 里面是不是已经分配了？

![Image](https://github.com/user-attachments/assets/29b41c62-eec9-4835-b97f-12cc0b03a204)

不是分配不了，是分配好了，程序刚启动时，容器内执行 nvidia-smi ，显示正常的，但是代码开始运行后，nvidia-smi 却提示需要使用超大的显存，这显然不对

**Comment by 2232729885** [comment]
> > Bot detected the issue body's language is not English, translate it automatically. 👯👭🏻🧑‍🤝‍🧑👫🧑🏿‍🤝‍🧑🏻👩🏾‍🤝‍👨🏿👬🏿
> 
> Reference this: [#1181](https://github.com/Project-HAMi/HAMi/issues/1181)
> 
> You can try upgrading to v2.6.1.

fiexed，good

**Comment by mesutoezdil** [comment]
Hi,

We are doing a cleanup of open issues to keep the backlog manageable. This issue was opened before 2026 and has not seen recent activity.

Please reply by 2026-07-13 to confirm this is still relevant and reproducible on the current version of HAMi. If we do not hear back, we will close this issue. You can always reopen it or file a new one if the problem persists.

Thank you for contributing to HAMi.
